include/envoy/network/connection_handler.h

```diff
@@ -37,12 +37,13 @@ class ConnectionHandler {
                               const Network::ListenerOptions& listener_options) PURE;
 
   /**
-   * Find a listener based on the provided listener port value.
-   * @param port supplies the port value
+   * Find a listener based on the provided listener address value.
+   * @param address supplies the address value.
    * @return a pointer to the listener or nullptr if not found.
    * Ownership of the listener is NOT transferred
    */
-  virtual Network::Listener* findListenerByPort(uint32_t port) PURE;
+  virtual Network::Listener*
+  findListenerByAddress(const Network::Address::InstancePtr& address) PURE;
```

Comment:
I tried, it does not work in testing file because no comparison of Address::Instance is defined.